source/loader/ur_adapter_registry.hpp

```diff
@@ -152,10 +152,123 @@ class AdapterRegistry {
         return paths.empty() ? std::nullopt : std::optional(paths);
     }
 
+    ur_result_t readPreFilterODS(std::string platformBackendName) {
+        // TODO: Refactor this to the common code such that both the prefilter and urDeviceGetSelected use the same functionality.
+        bool acceptLibrary = true;
```

Comment:
nit: I think this should be false by default, to cover `!*` case (e.g. `!*:level_zero`)

Reply:
This case was being handled, but I added a check that makes this case get handled faster and added a test for !*;level_zero